api/models.py

```diff
@@ -377,18 +383,14 @@ class Regression(Node):
         description='Type of the object',
         const=True
     )
-    regression_data: List[Node] = Field(
-        description='Regression details'
+
+    data: RegressionData = Field(
+        description="Regression details"
     )
 
     _OBJECT_ID_FIELDS = Node._OBJECT_ID_FIELDS + [
-        'regression_data.parent',
-    ]
-    _TIMESTAMP_FIELDS = Node._TIMESTAMP_FIELDS + [
-        'regression_data.created',
-        'regression_data.updated',
-        'regression_data.timeout',
-        'regression_data.holdoff',
+        'data.testrun_failed',
+        'data.testrun_passed',
```

Comment:
These are not timestamp fields. `_TIMESTAMP_FIELDS` are defined to translate timestamp fields for querying nodes.
Please see `Node._translate_timestamps` method for the implementation.

Reply:
Oh, sorry. I made a mistake while reading `diff`.
 
> First of all, what did the original implementation do?

When you send GET `/nodes` query with attributes, all the request query param values will be received as strings by default. And those values will be used to find matching nodes with `db.find_by_attributes`. See https://github.com/kernelci/kernelci-api/blob/main/api/main.py#L493.
It is not an issue when attributes are also string type in `Node` model such as `kind` or `name`. But when we send attributes having a different type such as `ObjectId` or `datetime`, a conversion from string to actual data type will be required in order to get items from DB.
e.g. `https://staging.kernelci.org:9000/latest/nodes?created=2023-07-17T11:52:31.913000`

That's when `Node._translate_object_ids` and `Node._translate_timestamps` come into the picture.

Reply:
`collection.find( )` works with an array of embedded documents with query filters like this.
Please see https://www.mongodb.com/docs/manual/tutorial/query-array-of-documents/

Also, you can experiment with it in your `mongo` shell.
For instance, you can try out:
```
db.regression.find({"regression_data.created": ISODate("2023-05-29T11:55:05.462Z")})
```